src/EnergyPlus/DXCoils.cc

```diff
@@ -2979,895 +2979,978 @@ void GetDXCoils(EnergyPlusData &state)
-            ErrorsFound = true;
-        }
-
-        thisDXCoil.RatedCOP(1) = Numbers(2);
-        if (thisDXCoil.RatedCOP(1) <= 0.0) {
-            ShowSevereError(state, format("{}{}=\"{}\", invalid", RoutineName, CurrentModuleObject, thisDXCoil.Name));
-            ShowContinueError(state, format("...{} must be > 0.0, entered value=[{:.2T}].", cNumericFields(2), Numbers(2)));
-            ErrorsFound = true;
-        }
-
-        thisDXCoil.RatedSHR(1) = Numbers(3);
-        if (thisDXCoil.RatedSHR(1) <= 0.0 || thisDXCoil.RatedSHR(1) > 1.0) {
-            ShowSevereError(state, format("{}{}=\"{}\", invalid", RoutineName, CurrentModuleObject, thisDXCoil.Name));
-            ShowContinueError(state, format("...{} must be > 0 and <= 1.  entered value=[{:.3T}].", cNumericFields(3), Numbers(3)));
-
-            ErrorsFound = true;
-        }
-
-        thisDXCoil.RatedInletDBTemp = Numbers(4);
-        if (thisDXCoil.RatedInletDBTemp <= 5.0) {
-            ShowSevereError(state, format("{}{}=\"{}\", invalid", RoutineName, CurrentModuleObject, thisDXCoil.Name));
-            ShowContinueError(state, format("...{} must be > 5 {{C}}.  entered value=[{:.1T}].", cNumericFields(4), Numbers(4)));
-            ErrorsFound = true;
-        }
-
-        thisDXCoil.RatedInletWBTemp = Numbers(5);
-        if (thisDXCoil.RatedInletWBTemp <= 5.0) {
-            ShowSevereError(state, format("{}{}=\"{}\", invalid", RoutineName, CurrentModuleObject, thisDXCoil.Name));
-            ShowContinueError(state, format("...{} must be > 5 {{C}}.  entered value=[{:.1T}].", cNumericFields(5), Numbers(5)));
-            ErrorsFound = true;
-        }
+            auto &thisDXCoil = state.dataDXCoils->DXCoil(DXCoilNum);
+            thisDXCoil.Name = Util::makeUPPER(thisObjectName);
+            thisDXCoil.DXCoilType = CurrentModuleObject;
+            thisDXCoil.DXCoilType_Num = HVAC::CoilDX_HeatPumpWaterHeaterPumped;
+            thisDXCoil.availSched = Sched::GetScheduleAlwaysOff(state); // heat pump water heater DX coil has no schedule
 
-        thisDXCoil.RatedInletWaterTemp = Numbers(6);
-        if (thisDXCoil.RatedInletWaterTemp <= 25.0) {
-            ShowSevereError(state, format("{}{}=\"{}\", invalid", RoutineName, CurrentModuleObject, thisDXCoil.Name));
-            ShowContinueError(state, format("...{} must be > 25 {{C}}.  entered value=[{:.1T}].", cNumericFields(6), Numbers(6)));
-            ErrorsFound = true;
-        }
+            // ErrorsFound will be set to True if problem was found, left untouched otherwise
+            VerifyUniqueCoilName(state, CurrentModuleObject, thisDXCoil.Name, ErrorsFound, CurrentModuleObject + " Name");
 
-        thisDXCoil.RatedAirVolFlowRate(1) = Numbers(7);
-        if (thisDXCoil.RatedAirVolFlowRate(1) != Constant::AutoCalculate) {
-            if (thisDXCoil.RatedAirVolFlowRate(1) <= 0.0) {
+            std::string const cRatedHeatingCapFieldName = "Rated Heating Capacity";
```

Comment:
This can be a `constexpr std::string_view`. There is no need to construct a dynamic string object at runtime to represent a constant string.

Reply:
Done.